src/main/java/io/anserini/collection/TwentynewsgroupsCollection.java

```diff
@@ -0,0 +1,146 @@
+/*
+ * Anserini: A Lucene toolkit for replicable information retrieval research
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.anserini.collection;
+
+import java.io.BufferedReader;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.nio.file.Path;
+import java.util.NoSuchElementException;
+
+/**
+ * An instance of the <a href="http://people.csail.mit.edu/jrennie/20Newsgroups">20 Newsgroup by date</a>.
+ * The collection contains 18846 documents, 20 different news sets,each corresponding to a different topic. 
+ * stored in plain text format. The collection is 14MB compressed, 35.9MB uncompressed.
+ */
+public class TwentynewsgroupsCollection extends DocumentCollection<TwentynewsgroupsCollection.Document> {
+
+  public TwentynewsgroupsCollection(Path path){
+    this.path = path;
+  }
+
+  @Override
+  public FileSegment<TwentynewsgroupsCollection.Document> createFileSegment(Path p) throws IOException {
+    return new Segment(p);
+  }
+
+  /**
+   * A file containing one document from the <a href="http://people.csail.mit.edu/jrennie/20Newsgroups">20 Newsgroup by date</a>.
+   * The file Name which is the Id of that document.
+   */
+  public static class Segment extends FileSegment<Document> {
+    private String fileName;
+    private String ID;
+
+    public Segment(Path path) throws IOException {
+      super(path);
+      this.fileName = path.toString();
+      String[] str_path = fileName.split("/");
+      this.ID = str_path[str_path.length-1];
+      this.bufferedReader = new BufferedReader(new InputStreamReader(new FileInputStream(this.fileName), "utf-8"));
+    }
+
+    @Override
+    public void readNext() throws IOException {
+      String record;
+      String From = "";
+      String Subject = "";
+      String Keywords = "";
+      String Organization = "";
+      String s = "";
+      while ((record = bufferedReader.readLine())!=null){
```

Comment:
code formatting; empty line above; two space indent, space before 'else'

Reply:
Sorry, I will be more careful on code formatting in the future. Fixed all code formatting/renaming problems in this collection now.